modules/catalog-dsl/src/main/java/org/apache/ignite/internal/catalog/sql/CreateFromAnnotationsImpl.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.catalog.sql;
+
+import static org.apache.ignite.catalog.ColumnSorted.column;
+import static org.apache.ignite.internal.catalog.sql.QueryUtils.mapArrayToList;
+import static org.apache.ignite.table.mapper.Mapper.nativelySupported;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.ignite.catalog.ColumnSorted;
+import org.apache.ignite.catalog.ColumnType;
+import org.apache.ignite.catalog.DefaultZone;
+import org.apache.ignite.catalog.IndexType;
+import org.apache.ignite.catalog.Options;
+import org.apache.ignite.catalog.annotations.Col;
+import org.apache.ignite.catalog.annotations.Column;
+import org.apache.ignite.catalog.annotations.Id;
+import org.apache.ignite.catalog.annotations.Index;
+import org.apache.ignite.catalog.annotations.Table;
+import org.apache.ignite.catalog.annotations.Zone;
+import org.apache.ignite.sql.IgniteSql;
+
+class CreateFromAnnotationsImpl extends AbstractCatalogQuery {
+    private CreateZoneImpl createZone;
+
+    private CreateTableImpl createTable;
+
+    private IndexType pkType;
+
+    CreateFromAnnotationsImpl(IgniteSql sql, Options options) {
+        super(sql, options);
+    }
+
+    CreateFromAnnotationsImpl keyValueView(Class<?> keyClass, Class<?> valueClass) {
+        if (keyClass.getAnnotation(Table.class) == null && valueClass.getAnnotation(Table.class) == null) {
+            throw new IllegalArgumentException("Table annotation must be present.");
```

Comment:
I think a user-facing error message should be as clear as possible. Here and everywhere else I would expect something like: "Can not find @Table annotation on keyCless.name() or valueClass.name. One of these classes must be annotated with @Table ...".